cli/internal/dataplane/write.go

```diff
@@ -251,6 +280,16 @@ func Write(ctx context.Context, uri *url.URL, inputReader io.Reader, options ...
 	return nil
 }
 
+func trackBuffer(flushInterval time.Duration, buffer *[]byte, bytesRead *int, inputReader io.Reader, err *error) {
+	for {
+		for start := time.Now(); time.Since(start) < flushInterval; {
+			if *bytesRead == 0 {
```

Comment:
reading wont happen again until the buffer is flushed which would cause delays. we could always create a new buffer and write it the moment it returns instead of using the same buffer from the main loop?